src/ripple/rpc/impl/LookupLedger.cpp

```diff
@@ -47,35 +64,58 @@ Status ledgerFromRequest (
 
     if (!hashValue.empty())
     {
+        if (! hashValue.isString ())
+            return {rpcINVALID_PARAMS, "ledgerHashNotString"};
+
         uint256 ledgerHash;
-        if (hashValue.isString() && ledgerHash.SetHex (hashValue.asString ()))
-            ledger = netOps.getLedgerByHash (ledgerHash);
-        else
+        if (! ledgerHash.SetHex (hashValue.asString ()))
             return {rpcINVALID_PARAMS, "ledgerHashMalformed"};
+           
+        if ((ledger = netOps.getLedgerByHash (ledgerHash)) == nullptr)
```

Comment:
Oh, I missed this initially - that ledger was being assigned!

consider putting the assignment to ledger as a separate line above and then just saying `if (ledger)`?  variable assignment as a side-effect of an expression is a little unexpected...

(This is distinct from variable _definition_ in an expression, like `for (auto i = x.begin()...`, which is a standard C++ construct...)